systemtest/src/test/java/io/strimzi/systemtest/kafka/TieredStorageST.java

```diff
@@ -138,10 +148,49 @@ void testTieredStorageWithAivenPlugin() {
         resourceManager.createResourceWithWait(clients.producerStrimzi());
 
         SetupMinio.waitForDataInMinio(suiteStorage.getNamespaceName(), BUCKET_NAME);
+
+        // Create admin-client to check offsets
+        resourceManager.createResourceWithWait(
+            AdminClientTemplates.plainAdminClient(
+                testStorage.getNamespaceName(),
+                testStorage.getAdminName(),
+                KafkaResources.plainBootstrapAddress(testStorage.getClusterName())
+            )
+            .editSpec()
+                .editOrNewTemplate()
+                    .editSpec()
+                        .editFirstContainer()
+                            // TODO - use latest test-clients after https://github.com/strimzi/test-clients/pull/103
+                            .withImage("quay.io/jstejska/test-clients:admin-kafka-3.7.1")
+                        .endContainer()
+                    .endSpec()
+                .endTemplate()
+            .endSpec().build()
+        );
+        final AdminClient adminClient = AdminClientUtils.getConfiguredAdminClient(testStorage.getNamespaceName(), testStorage.getAdminName());
+        // Fetch latest offsets - number of sent messages
+        String offsetData = adminClient.fetchOffsets(testStorage.getTopicName(), "latest");
+        long latestOffset = AdminClientUtils.getPartitionsOffset(offsetData, "0");
+
+        // Fetch earliest-local offsets - number of messages stored locally
+        offsetData = adminClient.fetchOffsets(testStorage.getTopicName(), "earliest-local");
+        long earliestLocalOffset = AdminClientUtils.getPartitionsOffset(offsetData, "0");
+        // Check that data are not present locally, earliest-local offset should be lower than latest offset
+        // latest should be 10000
+        // earliest-local should be lower than 10000
+        assertThat(earliestLocalOffset, lessThan(latestOffset));
+
         ClientUtils.waitForInstantProducerClientSuccess(testStorage);
```

Comment:
dumb question 2: We've sent 10000 messages above, what does this producer client do here?

Reply:
Sending messages is assynchronous process - `resourceManager.createResourceWithWait(clients.producerStrimzi());` just create job in kube that will start sending the messages. `ClientUtils.waitForInstantProducerClientSuccess(testStorage);` basically wait for success of the job. In case the job end in failure state, the test will fail